pagopa/io-eudiw-app
# feat: [EUDIW-176] Add application pin verification

## Short description

> [!NOTE]
> This implementation currently doesn't have an anti-tamper system, thus there's no limit in the number of attempts you can make.
>
> Depends on #48 

> [!CAUTION]
> Due to migrations not being handled at this stage, an app reinstallation might be required.

This PR adds the the pin verification flow when opening the app and when required by sensitive operations like adding a credential.

## List of changes proposed in this pull request

- Add the `identification` reducer which is needed to manage the identification state. Dispatching the `setIdentificationStarted` shows the identification modal which then dispatches the success or failure event;
- Add `IdentificationModal` which uses the `identification` reducer to handle the identification process via pin or biometric;
- Add the required biometric utility functions to perform a biometric identification; 
- Handle the reset of the application in different reducers where the state must be reset when preferences are reset due to a forgotten pin. This also restarts the setup saga;
- Rewrite startup login with two more states which are `WAIT_IDENTIFICATION` when the onboarding is complete and an identification modal should be shown before finishing the startup. `WAIT_ONBOARDING` when the onboarding must be completed before finishing the startup;
- Add identification before storing the pid;
- Add locales;

## How to test
Start the app and test the identification modal. Try to store a PID and the identification modal should pop-up. Also try to reset the pin.
